Objects/genericaliasobject.c

```diff
@@ -180,18 +180,31 @@ tuple_extend(PyObject **dst, Py_ssize_t dstindex,
 PyObject *
 _Py_make_parameters(PyObject *args)
 {
-    Py_ssize_t nargs = PyTuple_GET_SIZE(args);
+    Py_ssize_t nargs = PySequence_Length(args);
     Py_ssize_t len = nargs;
     PyObject *parameters = PyTuple_New(len);
     if (parameters == NULL)
         return NULL;
     Py_ssize_t iparam = 0;
     for (Py_ssize_t iarg = 0; iarg < nargs; iarg++) {
-        PyObject *t = PyTuple_GET_ITEM(args, iarg);
+        PyObject *t = PySequence_GetItem(args, iarg);
         // We don't want __parameters__ descriptor of a bare Python class.
         if (PyType_Check(t)) {
             continue;
         }
+        // Recursively call _Py_make_parameters for lists/tuples and
+        // add the results to the current parameters.
+        if (PyTuple_Check(t) || PyList_Check(t)) {
+            PyObject *subargs = _Py_make_parameters(t);
+            if (subargs == NULL) {
+                Py_DECREF(parameters);
+                return NULL;
+            }
+            iparam += tuple_extend(&parameters, iparam, &PyTuple_GET_ITEM(subargs, 0),
```

Comment:
`&PyTuple_GET_ITEM` looks a bit suspicious, could you add a test where the subargs contains more than one element?